core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/constraints/EnvelopeConstraint.java

```diff
@@ -34,8 +33,9 @@ public boolean initCheck(double direction, double position, double speed) {
         cursor = new EnvelopeCursor(envelope, direction < 0, partIndex, stepIndex, position);
         if (type == CEILING)
             return envelopeSpeed >= speed;
-        else
+        if (type == FLOOR)
             return envelopeSpeed <= speed;
+        return true;
```

Comment:
At this point a `switch` would probably be better, the third enum value would becomes explicit. It's not clear when reading this code that the last `return` is only reached if `type == MAINTAIN_SPEED`

Always returning `true` with a certain type of envelope seems weird. Is there a reason?

Reply:
The `true` default value is when no constraint type has been sent, which does not happen. It's just that calling this method without a constraint type doesn't really make sense, because without constraint the check is always `true`.

Reply:
I don't think there can be a case where no constraint type has been set. It seems like it's a fix to the "not every path returns something" error, which hides the problem when a new enum value is added. This can also be fixed with a switch.